instrumentor/controllers/instrumentationdevice/pods_webhook.go

```diff
@@ -70,12 +70,22 @@ func (p *PodsWebhook) getServiceNameForEnv(ctx context.Context, pod *corev1.Pod)
 	}
 	podWorkload.Namespace = req.Namespace
 
-	workloadObj, err := workload.GetWorkloadObject(ctx, client.ObjectKey{Namespace: podWorkload.Namespace, Name: podWorkload.Name}, podWorkload.Kind, p.Client)
+	workloadObj := workload.ClientObjectFromWorkloadKind(podWorkload.Kind)
+	err = p.Client.Get(ctx, client.ObjectKey{Namespace: podWorkload.Namespace, Name: podWorkload.Name}, workloadObj)
 	if err != nil {
-		logger.Error(err, "failed to get workload object from cache. cannot check for workload annotation. using workload name as OTEL_SERVICE_NAME")
+		logger.Error(err, "failed to get workload object from cache. cannot check for workload source. using workload name as OTEL_SERVICE_NAME")
 		return &podWorkload.Name, podWorkload
 	}
-	resolvedServiceName := workload.ExtractServiceNameFromAnnotations(workloadObj.GetAnnotations(), podWorkload.Name)
+
+	resolvedServiceName, err := sourceutils.OtelServiceNameBySource(ctx, p.Client, workloadObj)
+	if err != nil {
+		logger.Error(err, "failed to get OTel service name from source. using workload name as OTEL_SERVICE_NAME")
+		return &podWorkload.Name, podWorkload
```

Comment:
nit: instead of return, set `resolvedServiceName` so the function has better control structure